recognition/include/pcl/recognition/face_detection/face_common.h

```diff
@@ -151,14 +157,15 @@ namespace pcl
             for (std::size_t j = 0; j < 3; j++)
               stream.read (reinterpret_cast<char*> (&covariance_rot_ (i, j)), sizeof(covariance_rot_ (i, j)));
 
+          if (num_of_sub_nodes <= 0)
+          {
+            PCL_THROW_EXCEPTION (pcl::IOException, "Error! Number of sub nodes read from file is not positive!");
```

Comment:
It might not be good to throw in every case. In some cases, 0 might be an acceptable value

Reply:
Is it? Because the original code had a check `num_of_sub_nodes > 0`

Reply:
A check yes, but not a throw. A throw exits immediately, halting all code (till it's caught). Checking and skipping is quite different